pkg/controller/backup_controller.go

```diff
@@ -397,10 +397,16 @@ func (c *backupController) prepareBackupRequest(backup *velerov1api.Backup) *pkg
 // - a given provider's default location name is added to .spec.volumeSnapshotLocations if one
 //   is not explicitly specified for the provider (if there's only one location for the provider,
 //   it will automatically be used)
+// if backup has snapshotVolume disabled then it returns empty VSL
 func (c *backupController) validateAndGetSnapshotLocations(backup *velerov1api.Backup) (map[string]*velerov1api.VolumeSnapshotLocation, []string) {
 	errors := []string{}
 	providerLocations := make(map[string]*velerov1api.VolumeSnapshotLocation)
 
+	// if snapshotVolume is set to false then we don't need to validate volumesnapshotlocation
+	if backup.Spec.SnapshotVolumes != nil && !*backup.Spec.SnapshotVolumes {
```

Comment:
We have some helper functions at https://github.com/vmware-tanzu/velero/blob/master/pkg/util/boolptr/boolptr.go to handle this.

Reply:
done